File: fatorialReverso.c
/**
 * 
 * autor Mauricio Freire
 * Idem a versão em Python.
 * Idem Python version.
*/

#include <stdio.h>

int rev_fat(int);

int main()
{
    int f, r;
    printf("Digite um numero: ");
    scanf("%d", &f);
    
    r = rev_fat(f);
    if (!r)
        printf("%d nao eh produto fatorial de nenhum numero!\n", f);
    else
        printf("%d eh produto fatorial de %d\n", f, r);
    return 0;
}

int rev_fat(int fat)
{
    int v = fat, i;
    for (i = 1; i < v; i++) {
        fat /= i;
        if (fat == 1) return i;
    }
    return 0;
}

File: fatorialReverso.java
# Idem a versão em Python.
# Idem to Python version.

class rev_Fat {
    private int v;
    
    int reverseFatorial(int fat) {
        v = fat;
        
        for (int i = 1; i < v; i++)
            while (fat > i) {
                System.out.println(fat + "" + i);
                fat /= i;
                if (fat == 1) return i;
            }
        return 0;
    }
}

public class Main {
    public static void main(String args[]) {
        rev_Fat ob = new rev_Fat();
        int f = 120, r;
        
        r = ob.reverseFatorial(f);
        
        if (r == 0)
            System.out.println(f + " não é fatorial de nenhum número!");
        else
            System.out.println(f + " é fatorial de " + r);
    }
}
